### PR TITLE
[BW-44] Work with favorites screen

### PR DESCRIPTION
**Pull request by FesenkoG**

### CODE REVIEW

File: BankexWallet/InterfaceElements/DesignElements/Favourites/FavoritesHelperViewController.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+class FavoritesHelperViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    @IBOutlet weak var tableView: UITableView!
+    var allFavorites: [FavoriteModel]?
+    weak var delegate: CellSelectedProtocol?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.delegate = self
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/Favourites/FavoritesHelperViewController.swift]: Can be set in storyboard

File: BankexWallet/InterfaceElements/DesignElements/Favourites/FavoritesHelperTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import UIKit
+
+class FavoritesHelperTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var profileImgView: UIImageView!
+    @IBOutlet weak var addressLbl: UILabel!
+    @IBOutlet weak var shortNameLbl: UILabel!
+    @IBOutlet weak var nameLbl: UILabel!
+    override func awakeFromNib() {
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/Favourites/FavoritesHelperTableViewCell.swift]: Delete the code you don't really use

File: BankexWallet/InterfaceElements/DesignElements/Favourites/FavoritesHelperViewController.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        tableView.dataSource = self
+        
+    }
+
+    
+    @IBAction func dismissView(_ sender: UIGestureRecognizer) {
+        self.performSegue(withIdentifier: "unwindToSend", sender: nil)
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/Favourites/FavoritesHelperViewController.swift]: In this case implementation of this method is useless

File: BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/CustomSegues.swift [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+//
+
+import UIKit
+
+class ShowFavoritesSegue: UIStoryboardSegue {
+    override func perform() {
+        transition()
+    }
+    
+    func transition() {
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/CustomSegues.swift]: Too many magic numbers in this method. You should either put them into constants, or get them from view itself

File: BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift [hunk 5]
```diff
@@ -97,6 +102,8 @@ Retriable {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         enterAddressTextfield.text = selectedFavoriteAddress ?? enterAddressTextfield.text
+        favorites = favService?.getAllStoredAddresses()
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift]: Why do you need to do it here, not in the place you'll actually use it?

File: BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift [hunk 6]
```diff
@@ -127,6 +134,10 @@ Retriable {
     
     var sendingProcess: SendingResultInformation?
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let favVC = segue.destination as? FavoritesHelperViewController {
+            favVC.allFavorites = self.favorites
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift]: Why should you keep all the array in current controller, if you use it only in another one?

File: BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift [hunk 7]
```diff
@@ -203,13 +214,16 @@ Retriable {
     
     @IBAction func addFromFavouritesTapped(_ sender: Any) {
         // TODO: Open favs list
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift]: Seems weird for now 😄

File: BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift [hunk 8]
```diff
@@ -203,13 +214,16 @@ Retriable {
...
+        
+        
     }
     
     @IBAction func insertFromClipboardTapped(_ sender: Any) {
         enterAddressTextfield.text = UIPasteboard.general.string
     }
     
     @IBAction func deleteSelectedFavTapped(_ sender: Any) {
+
```
Comment (K0rka) [inline BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift]: And what about implementation of this method?

File: BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift (hunk 6 above)
Comment (FesenkoG) [reply BankexWallet/InterfaceElements/DesignElements/SendTokenViewController/SendTokenViewController.swift]: This is not the only usage of that array in the current controller, another one is in textfield's delegate method

**Pull request closed without merging**